File: frontend/src/contracts/kioskDiscovery.ts
export interface DiscoveryRequest {
    available_time: number;
    interest_tags: string[];
    district_id: string;
}

export interface TransportHubOption {
    name: string;
    type: string; // 'BUS_STAND' | 'RAILWAY_STATION' | 'TAXI_POINT' | 'FERRY' | 'OTHER'
    notes?: string;
}

export interface DiscoveryResult {
    hotspot_id: string;
    title: string;
    short_description: string;
    district_name: string;
    cluster_label: string;
    typical_duration_min?: number;
    distance_band: string; // 'NEAR' | 'MEDIUM' | 'FAR'
    approx_travel_time_min?: number;
    nearest_transport_hub?: TransportHubOption;
    safety_note?: string;
}

export interface DistrictOption {
    district_id: string;
    name: string;
}

File: backend/shared/contracts/kiosk_discovery.py
from dataclasses import dataclass, field
from typing import List, Optional

@dataclass
class DiscoveryRequest:
    available_time: int
    interest_tags: List[str]
    district_id: Optional[str] = None

@dataclass
class TransportHubOption:
    name: str
    type: str # 'BUS', 'RAIL', 'TAXI', 'FERRY', 'OTHER'
    notes: Optional[str] = None

@dataclass
class DiscoveryResult:
    hotspot_id: str
    title: str
    short_description: str
    district_name: str
    cluster_label: str
    typical_duration_min: Optional[int]
    distance_band: str # 'NEAR', 'MEDIUM', 'FAR'
    approx_travel_time_min: Optional[int]
    nearest_transport_hub: Optional[TransportHubOption]
    safety_note: Optional[str] = None
